VCF header fixes (A -> R); all allele annotations

chapmanb opened the pull request:
Fixes issue identified by @kgururaj when investigating GenomicsDB
dropping calls (https://github.com/broadinstitute/gatk/issues/3429#issuecomment-324188028)
which is due to incorrect VCF header length descriptions.

It looks like this mismatch was reported early by @LeeTL1220
in #3296 when validating VCFs.

@davidbenjamin provided a partial fix in #3351, generalizing the output
to include the option of specifying R instead of A using
`includeRefAllele`, fixing MFRL and removing MCL.

This fixes the 3 remaining cases, MMQ (which breaks the GenomicsDB
example), MPOS and MBQ.

Fixes #3296
Fixes #3429

lbergelson commented:
@davidbenjamin Could you take a look at this one, it seems like the changes fall in your code.

chapmanb commented:
Apologies, I'd missed the unit test before. We expect these to change since we now include REF alleles in the outputs. I updated the tests and rebased so hopefully checks will now finish cleanly.

chapmanb commented:
David -- thanks again for the review. I tested the latest head and your annotation fixes in #3351 do resolve the issue with incompatible headers, resulting in correct behavior with GenomicsDB. There is no need for this patch so closing -- a new release should fix with your improvements should fix that issue. Thanks again.
